src/components/application/index.js

```diff
@@ -1,23 +1,365 @@
+            gradYear: {
+                required: true,
+                enum: gradYears,
+            },
+            numPrevHackathons: {
+                validate: {
+                    fn: validation.isNumber,
+                    message: 'Must be a number',
+                },
+            },
+            personalWebsite: {
+                validate: {
+                    fn: validation.isWebsite,
+                    message: 'Must be a valid website',
+                },
+            },
+            githubUsername: {
+                validate: {
+                    fn: validation.isString,
+                    message: 'Must be a valid string',
+                },
+            },
+            resume: {
+                type: 'file',
+            }
+        };
+
+        this.DEFAULT_ERRORS = {
+            firstName: {
+                highlight: false,
+                message: null,
+            },
+            lastName: {
+                highlight: false,
+                message: null,
+            },
+            age: {
+                highlight: false,
+                message: null,
+            },
+            ethnicity: {
+                highlight: false,
+                message: null,
+            },
+            school: {
+                highlight: false,
+                message: null,
+            },
+            major: {
+                highlight: false,
+                message: null,
+            },
+            gradYear: {
+                highlight: false,
+                message: null,
+            },
+            numPrevHackathons: {
+                highlight: false,
+                message: null,
+            },
+            personalWebsite: {
+                highlight: false,
+                message: null,
+            },
+            githubUsername: {
+                highlight: false,
+                message: null,
+            },
+            resume: {
+                highlight: false,
+                message: null,
+            },
+        };
+
         this.state = {
-            app: {},
+            app: {
+                firstName: '',
+                lastName: '',
+                age: '',
+                ethnicity: '',
+                school: null,
+                major: '',
+                gradYear: null,
+                numPrevHackathons: '',
+                personalWebsite: '',
+                githubUsername: '',
+                resume: null,
+            },
+            errors: this.DEFAULT_ERRORS,
         };
     }
 
+    componentWillReceiveProps(nextProps) {
+        // shallow check
+        if (nextProps.app !== this.props.app) {
+            console.log("yes", nextProps.app);
+        }
+    }
+
+    validateField(validation, value) {
+        let highlight = false, message = null;
+
+        if (validation.required) {
+            if (!value) {
+                highlight = true;
+                message = 'This field is required';
+                return {
```

Comment:
do these return statements need to be here, in any of the conditionals? since you're just returning the same object at the bottom anyway

Reply:
I'm returning it because I don't want errors that match multiple conditions to be overwritten. I want to return the first error that's being caught. And by default, it would return `{ highlight: false, message: null }`